libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -711,7 +722,13 @@ shrinkPred (Block ps) = TruePred : FalsePred [] : ps ++ map Block (shrinkList sh
 shrinkPred (Assert es t) = TruePred : FalsePred [] : [Assert [] t | not $ null es]
 shrinkPred (Explain _ p) = TruePred : FalsePred [] : [p]
 shrinkPred (When b p) = TruePred : FalsePred [] : p : map (When b) (shrinkPred p)
-shrinkPred (Exists k (x :-> p)) = TruePred : FalsePred [] : [Exists k (x :-> p') | p' <- shrinkPred p]
+-- NOTE: You can't shrink `Exists` because it might make the `Env -> a` invalid!
+-- e.g. you start with
+-- `constrained $ \ x -> exists (\eval -> pure $ eval x) $ \ y -> [ x ==. y, 10 <. y ]`
+-- and shrink it to
+-- `constrained $ \ x -> exists (\eval -> pure $ eval x) $ \ y -> [ 10 <. y ]`
+-- and suddenly the thing you're generating is BS w.r.t the checking!
+shrinkPred Exists {} = [TruePred, FalsePred []]
```

Comment:
Do you mean that the existentially quantified  variable "x" has disappeared?

Reply:
In the example the equality constraint has disappeared